### PR TITLE
feat: enhance /speckit:constitution with tier selection and constitution customization (task-244)

### PR DESCRIPTION
**Pull request by jpoley**
## Summary

- Enhances `/speckit:constitution` slash command to fulfill task-244 requirements
- Adds `--tier {light|medium|heavy}` flag support
- Auto-detects tier based on project complexity (0-15 point scoring algorithm)
- Customizes constitution template with detected tech stack
- Preserves NEEDS_VALIDATION markers for user review
- Outputs validation checklist after generation
- Maintains existing repo-facts.md functionality

## Changes

- **templates/commands/speckit/constitution.md**: Enhanced command template

### Features Added
- Tier auto-detection based on: language count, CI/CD, testing, security tools, containers
- Constitution template customization with placeholders replaced
- Validation checklist output
- Edge case handling (invalid tier, missing template, existing constitution)

## Test plan

- [x] All 2410 tests pass
- [x] Ruff format and lint checks pass
- [ ] CI pipeline passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: templates/commands/speckit/constitution.md [hunk 1]
```diff
@@ -296,13 +378,97 @@ This document contains automatically detected repository characteristics. LLM ag
...
+- `[REPORTING_WINDOW]` - Leave as NEEDS_VALIDATION placeholder
+
+#### 4.4 Perform Template Replacements
+
+Replace these placeholders in the template:
+
+1. `[PROJECT_NAME]` → Detected project name (from Step 4.2)
+2. `[LANGUAGES_AND_FRAMEWORKS]` → Formatted tech stack (from Step 4.3)
+3. `[LINTING_TOOLS]` → Linting tools list (medium/heavy only)
+4. `[CI_CD_TOOLS]` → CI/CD platform details (heavy only)
```
[the remark this answers is not in the document]
Comment (jpoley) [reply templates/commands/speckit/constitution.md]: having CICD is for all

**Comment by jpoley** [comment]
fixes, some included some needed
